framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -754,17 +756,47 @@ public AnnotationFormatter getAnnotationFormatter() {
      *
      * <p>
      * Subclasses cannot override this method; they should override
-     * {@link #createSupportedTypeQualifiers createSupportedTypeQualifiers} instead.
-
+     * {@link #createSupportedTypeQualifiers createSupportedTypeQualifiers}
+     * instead.
+     *
      * @see #createSupportedTypeQualifiers()
      *
-     * @return the type qualifiers supported this processor, or an empty
-     * set if none
+     * @return an immutable HashSet of the supported type qualifiers, or an
+     *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
         return supportedQuals;
     }
 
+    /**
+     * Defines alphabetical sort ordering for qualifiers
+     */
+    private static final Comparator<Class<? extends Annotation>> QUALIFIER_SORT_ORDERING
+    = new Comparator<Class<? extends Annotation>>() {
+        @Override
+        public int compare(Class<? extends Annotation> a1, Class<? extends Annotation> a2) {
+            return a1.getCanonicalName().compareTo(a2.getCanonicalName());
+        }
+    };
+
+    /**
+     * Returns an alphabetically sorted immutable set of the type qualifiers
+     * supported by this checker. This method is useful for debug printing
+     * purposes, but otherwise returns the same set as
+     * {@link #getSupportedTypeQualifiers()}.
+     *
+     * @see #getSupportedTypeQualifiers()
+     *
+     * @return a sorted and immutable TreeSet of the supported type qualifiers,
+     *         or an empty set if no qualifiers are supported
+     */
+    public final Set<Class<? extends Annotation>> getSortedSupportedTypeQualifiers() {
```

Comment:
It's weird that this method name is called getSORTED... when the return type is a Set, which by definition is not sorted.  Maybe it should return some sort of ordered collection rather than a Set.